lib/linguist/repository.rb

```diff
@@ -157,10 +159,9 @@ def compute_stats(old_commit_oid, cache = nil)
           blob = Linguist::LazyBlob.new(repository, delta.new_file[:oid], new, mode.to_s(8))
 
           # Skip vendored or generated blobs
-          next if blob.vendored? || blob.generated? || blob.language.nil?
+          next if blob.vendored? || blob.documentation? || blob.generated? || blob.language.nil?
 
-          # Only include programming languages and acceptable markup languages
-          if blob.language.type == :programming || Language.detectable_markup.include?(blob.language.name)
+          if DETECTABLE_TYPES.include?(blob.language.type)
```

Comment:
Would it be worth wrapping this and the conditionals above into `BlobHelper#include_in_languages_stats?` or something like that?

```ruby
module BlobHelper
  def include_in_language_stats?
    !vendored? &&
    !documentation? &&
    !generated? &&
    language && DETECTABLE_TYPES.include?(language.type)
  end
end
```